# Nitroglycerin
[![Circle CI](https://circleci.com/gh/Originate/nitroglycerin.svg?style=shield)](https://circleci.com/gh/Originate/nitroglycerin)
[![Dependency Status](https://david-dm.org/originate/nitroglycerin.svg)](https://david-dm.org/originate/nitroglycerin)
[![Coverage Status](https://coveralls.io/repos/Originate/nitroglycerin/badge.svg?branch=master&service=github)](https://coveralls.io/github/Originate/nitroglycerin?branch=master)

_Makes JavaScript functions so volatile and explosive that they immediately throw on errors._

This is mostly useful for testing and simple command-line scripts.
In production you most likely don't want to throw exceptions.


## How it works

Let's say you have a small piece of test code that needs to set something up:

```coffeescript
resetDatabase (err) ->
  if err then throw new Error(err)
  createUser name: 'Alfred Nobel', (err, alfred) ->
    if err then throw new Error(err)
    # run your tests using alfred here
```

Checking that these calls didn't error out requires a bit of ceremony in Node.js,
either in the form of manual error checking as described above, promises,
or [async.waterfall](https://github.com/caolan/async#waterfall).
Nitroglycerin removes most of this ceremony:

```coffeescript
N = require 'nitroglycerin'

resetDatabase N ->
  createUser name: 'Alfred Nobel', N (alfred) ->
    # run your tests using alfred here
```

The functions wrapped in Nitroglycerin
will immediately throw an exception when they receive an error
or execute the callback with the error argument stripped.


## Development

See the [developer documentation](CONTRIBUTING.md)
